### PR TITLE
Monitoring logging

### PR DESCRIPTION
**Pull request by izzyblues**
The logging is based on this changes to the go logger library: https://github.com/Financial-Times/go-logger/pull/5

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -76,8 +76,7 @@ func main() {
 		EnvVar: "PRODUCER_TOPIC",
 	})
 
-	log.SetFormatter(&log.JSONFormatter{TimestampFormat: time.RFC3339Nano})
-	log.SetLevel(log.InfoLevel)
+	log.InitDefaultLogger(appSystemCode)
 	log.Info("[Startup] pac-annotations-mapper is starting ")
```
Comment (peteclark-ft) [inline main.go:80]: Should this use the new `ServiceStarted` func?

File: service/messageHandler_test.go [hunk 2]
```diff
@@ -234,9 +247,21 @@ func TestPredicateValidation(t *testing.T) {
 	actualAnnotations := actualBody.Annotations
 	assert.Len(t, actualAnnotations, 1, "annotations")
 	assert.Equal(t, brandUuid, actualAnnotations[0].Concept.ID, "brand annotation")
+
+	// check of monitoring logging
```
Comment (peteclark-ft) [inline service/messageHandler_test.go:242]: Could these monitoring logging checks be refactored out?

File: main.go (hunk 1 above)
Comment (izzyblues) [reply main.go:80]: I used this [PR](https://github.com/Financial-Times/annotations-mapper/pull/2) as a reference for which kind of events we need to keep track. Even if the startup event is defined in the `go-logger` library, it is not a a kind of event that the monitoring system keeps track.

File: service/messageHandler_test.go (hunk 2 above)
Comment (izzyblues) [reply service/messageHandler_test.go:242]: Makes absolutely sense!

File: service/messageHandler.go [hunk 3]
```diff
@@ -73,18 +79,30 @@ func (mapper *AnnotationMapperService) HandleMessage(msg kafka.FTMessage) error
 
 	marshalledAnnotations, err := json.Marshal(mappedAnnotations)
 	if err != nil {
-		requestLog.Error("Error marshalling the concept annotations", err)
+		log.WithMonitoringEvent(mapperEvent, tid, annotationsContentType).
+			WithUUID(metadataPublishEvent.UUID).
+			WithValidFlag(true).
+			WithError(err).
```
Comment (galiri) [inline service/messageHandler.go]: what does the isValid refer to here?

File: service/messageHandler.go [hunk 4]
```diff
@@ -5,21 +5,24 @@ import (
...
 	"github.com/Financial-Times/kafka-client-go/kafka"
 	"github.com/satori/go.uuid"
-	log "github.com/sirupsen/logrus"
 )
 
 const messageTimestampDateFormat = "2006-01-02T15:04:05.000Z"
 
+const mapperEvent = "Map"
+const annotationsContentType = "annotations"
+
```
Comment (galiri) [inline service/messageHandler.go]: can you rename this annotationType or something like that  it's not a content type.

**Review by galiri — CHANGES_REQUESTED** [review]

File: service/messageHandler.go (hunk 4 above)
Comment (izzyblues) [reply service/messageHandler.go]: Fixed!

File: service/messageHandler.go (hunk 3 above)
Comment (izzyblues) [reply service/messageHandler.go]: It is explained [here](https://github.com/Financial-Times/go-logger/pull/5/files#diff-04c6e90faac2675aa89e2176d2eec7d8R26).
It says if the processed message is valid.

**Review by peteclark-ft — APPROVED** [review]

File: vendor/vendor.json [hunk 5]
```diff
@@ -3,14 +3,30 @@
...
+		},
+		{
+			"checksumSHA1": "eQJNFoxeNYlvXsrc762OCAmvXCE=",
+			"path": "github.com/Financial-Times/go-logger/test",
+			"revision": "fb86ab056054cb9b58d295bd2ffbeac0123900e5",
+			"revisionTime": "2017-10-16T13:51:52Z",
+			"version": "0.0.3-custom-json-formatter-rc.5",
+			"versionExact": "0.0.3-custom-json-formatter-rc.5"
+		},
+		{
```
Comment (kghatton) [inline vendor/vendor.json]: Needs a release version before this can be ~approved~ merged.

**Review by galiri — COMMENTED** [review]
approved subject to release version for loggig library

**Review by kghatton — APPROVED** [review]

**Review by galiri — APPROVED** [review]

**Pull request merged**